microbench/db_basic_bench.cc

```diff
@@ -129,6 +246,1019 @@ static void DBClose(benchmark::State& state) {
+  state.counters["get_from_output_files_time"] = benchmark::Counter(
+      get_from_output_files_time, benchmark::Counter::kAvgIterations);
+  state.counters["new_table_block_iter_nanos"] = benchmark::Counter(
+      new_table_block_iter_nanos, benchmark::Counter::kAvgIterations);
+  state.counters["block_seek_nanos"] =
+      benchmark::Counter(block_seek_nanos, benchmark::Counter::kAvgIterations);
+  state.counters["get_cpu_nanos"] =
+      benchmark::Counter(get_cpu_nanos, benchmark::Counter::kAvgIterations);
+  state.counters["get_from_table_nanos"] = benchmark::Counter(
+      get_from_table_nanos, benchmark::Counter::kAvgIterations);
+
+  if (state.thread_index == 0) {
+    char min_buff[256], max_buff[256];
+    const Range r(kg.MinKey(min_buff), kg.MaxKey(max_buff));
+    uint64_t size;
+    auto s = db->GetApproximateSizes(&r, 1, &size);
+    if (!s.ok()) {
+      state.SkipWithError(s.ToString().c_str());
+    }
+    state.counters["db_size"] = size;
+
+    s = db->Close();
+    if (!s.ok()) {
+      state.SkipWithError(s.ToString().c_str());
+    }
+    DestroyDB(db_name, options);
+  }
+}
+
+BENCHMARK(SimpleGetWithPerfContext)->Iterations(1000000);
+
+std::string GenerateKey(int primary_key, int secondary_key, int padding_size,
+                        Random* rnd) {
+  char buf[50];
+  char* p = &buf[0];
+  snprintf(buf, sizeof(buf), "%6d%4d", primary_key, secondary_key);
+  std::string k(p);
+  if (padding_size) {
+    k += rnd->RandomString(padding_size);
+  }
+
+  return k;
+}
+
+void GenerateRandomKVs(std::vector<std::string>* keys,
+                       std::vector<std::string>* values, const int from,
+                       const int len, const int step = 1,
+                       const int padding_size = 0,
+                       const int keys_share_prefix = 1) {
+  Random rnd(302);
+
+  // generate different prefix
+  for (int i = from; i < from + len; i += step) {
+    // generating keys that shares the prefix
+    for (int j = 0; j < keys_share_prefix; ++j) {
+      keys->emplace_back(GenerateKey(i, j, padding_size, &rnd));
+      // 100 bytes values
+      values->emplace_back(rnd.RandomString(100));
+    }
+  }
+}
+
+static void DataBlockSeek(benchmark::State& state) {
+  Random rnd(301);
+  Options options = Options();
+
+  BlockBuilder builder(16, true, false,
+                       BlockBasedTableOptions::kDataBlockBinarySearch);
+
+  int num_records = 500;
+  std::vector<std::string> keys;
+  std::vector<std::string> values;
+
+  GenerateRandomKVs(&keys, &values, 0, num_records);
+
+  for (int i = 0; i < num_records; i++) {
+    std::string ukey(keys[i] + "1");
+    InternalKey ikey(ukey, 0, kTypeValue);
+    builder.Add(ikey.Encode().ToString(), values[i]);
+  }
+
+  Slice rawblock = builder.Finish();
+
+  BlockContents contents;
+  contents.data = rawblock;
+  Block reader(std::move(contents));
+  const InternalKeyComparator icmp(BytewiseComparator());
```

Comment:
Is `BytesiseComparator()` intended to be hard-coded?

Reply:
good point, removed that and use `options.comparator` instead.